include/odp/api/spec/pool.h

```diff
@@ -173,6 +173,107 @@ int odp_pool_stats(odp_pool_t pool, odp_pool_stats_t *stats);
+ * @param param    odp_pool_ext_param_t to be initialized
+ */
+void odp_pool_ext_param_init(odp_pool_type_t type, odp_pool_ext_param_t *param);
+
+/**
+ * Create an external memory pool
+ *
+ * This routine is used to create a pool. The use of pool name is optional.
+ * Unique names are not required. However, odp_pool_lookup() returns only a
+ * single matching pool. Use odp_pool_ext_param_init() to initialize parameters
+ * into their default values.
+ *
+ * @param name     Name of the pool or NULL. Maximum string length is ODP_POOL_NAME_LEN.
+ * @param param    Pool parameters
+ *
+ * @return Pool handle on success
+ * @retval ODP_POOL_INVALID on failure
+ */
+odp_pool_t odp_pool_ext_create(const char *name, const odp_pool_ext_param_t *param);
+
+/**
+ * Populate external memory pool with buffer memory
+ *
+ * Populate can be called multiple times to add memory buffers into the pool. Application must
+ * populate the pool with the exact number of buffers specified in pool parameters. The pool is
+ * ready to be used for allocations only after all populate calls have returned successfully.
+ * Application marks the last populate call with ODP_POOL_POPULATE_DONE flag.
+ *
+ * Depending on pool usage (and ODP implementation), the memory may need to be accessible by
+ * HW accelerators. Application may use e.g. odp_shm_reserve() with ODP_SHM_HW_ACCESS flag to
+ * ensure HW access. The memory area used for one pool (from the lowest addressed buffer to
+ * the end of the highest addressed buffer) must not overlap with the memory area used for
+ * any other pool.
+ *
+ * Pool type defines memory buffer layout and where the buffer pointer (buf[N]) points
+ * in the layout. Pool capabilities specify requirements for buffer size, layout and
+ * pointer alignment.
+ *
+ * For packet pools, packet buffer layout is shown below. The packet headroom (odp_packet_head())
+ * starts immediately after the application header. For a segmented packet, each segment has this
+ * same layout. Buffer size includes all headers, headroom, data, tailroom and trailer.
+ *
+ * @code{.unparsed}
+ *
+ *         buf[N] --->  +-------------------------------+    --                     --
+ *                      |                               |     |                      |
+ *                      | ODP header (optional)         |      > odp_header_size     |
+ *                      |                               |     |                      |
+ *                      +-------------------------------+    --                      |
+ *                      |                               |     |                      |
+ *                      | Application header (optional) |      > app_header_size     |
+ *                      |                               |     |                       > buf_size
+ *                      +-------------------------------+    --                      |
+ * odp_packet_head()--> |                               |                            |
```

Comment:
Minor comment. odp_packet_head() should be in line 249.

Reply:
To me +-----------+ line does not contain any bytes, it's just a line in the drawing dividing e.g. header fields, or header from payload. See e.g. how RFCs document headers in ascii graphics https://datatracker.ietf.org/doc/html/rfc791

In the figure, odp_packet_head() points now to the first byte after application header.

Reply:
Okay.

In that case does **buf[N] --->** also need to change ?

Reply:
Yes, I'll move that.